modules/gplazma2-pyscript/src/main/java/org/dcache/gplazma/pyscript/PyscriptPlugin.java

```diff
@@ -42,6 +43,35 @@ private static List<Path> listDirectoryContents(String directoryPath) throws IOE
         return Files.list(Path.of(directoryPath)).toList();
     }
 
+    private void executePythonFile(Path path) throws IOException {
+        String content = Files.lines(path).collect(
+                Collectors.joining("\n")
+        );
+        // execute file content, loading the function "py_auth" into the namespace
+        PI.exec(content);
+    }
+
+    private static List<String> principalSetToStringList(Set<Principal> principals) {
+        if (principals.isEmpty()) {
+            // empty set handling
+            return new ArrayList<String>();
+        } else {
+            String principalsString = Subjects.toString(
+                    Subjects.ofPrincipals(principals)
+            );
+            String[] principalsArray = principalsString.substring(
```

Comment:
This looks quite creative :). I suggest adding a special method in the `Subjects` class that does this (with a bit of copy-paste from existing `Subjects#toString`.